### PR TITLE
Add support for multiple Cesium ion servers

### PR DESCRIPTION
**Pull request by kring**
This is the Unity equivalent of https://github.com/CesiumGS/cesium-unreal/pull/1309.

### CODE REVIEW

**Comment by joseph-kaile** [comment]
Works well, one suggestion:

It would be nice to have the host url in the error message as well:

Exception: Request failed: Cannot resolve destination host
UnityEngine.Debug:LogError (object)
Reinterop.ReinteropInitializer:UnityEngine_Debug_CallLogError_FA05wu8x__otZNsgdHTnU9A (intptr) (at ./Reinterop/Reinterop.RoslynSourceGenerator/ReinteropInitializer.cs:3023)
CesiumForUnity.CesiumIonSession:Tick () (at ./Reinterop/Reinterop.RoslynSourceGenerator/CesiumIonSession-generated.cs:233)
CesiumForUnity.CesiumEditorUtility:UpdateIonSession () (at ./Packages/com.cesium.unity/Editor/CesiumEditorUtility.cs:26)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions ()

File: Runtime/CesiumIonServer.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#endif
+
+                return result;
+            }
+        }
+
+        /// <summary>
+        /// Gets the current Cesium ion server that should be assigned to newly-created objects.
+        /// </summary>
+        public static CesiumIonServer currentForNewObjects
```
Comment (j9liu) [inline Runtime/CesiumIonServer.cs:108]: For consistency with the Unreal PR

```suggestion
        public static CesiumIonServer serverForNewObjects
```

**Review by j9liu — COMMENTED** [review]
Looks great @kring ! I just had a few small comments.

File: Editor/CesiumEditorWindow.cs [hunk 2]
```diff
@@ -11,7 +13,7 @@ public partial class CesiumEditorWindow : EditorWindow
     {
         public static CesiumEditorWindow currentWindow = null;
 
-        [MenuItem("Cesium/Cesium")]
+        [MenuItem("Tools/Cesium/Cesium")]
```
[the remark this answers is not in the document]
Comment (kring) [reply Editor/CesiumEditorWindow.cs:16]: You weren't supposed to notice that! 😛
Just kidding, sorry I didn't mention it explicitly. The top-level Cesium menu has always bothered me a bit. I'm not sure that Tools is right, either, though. Maybe Window would be best? I guess I'll just revert it for now. There's certainly no good reason it has to be part of this PR.

[the remark this answers is not in the document]
Comment (j9liu) [reply Editor/CesiumEditorWindow.cs]: 😆 No worries -- I'm not opposed to moving it if there's a more proper place! But it should probably be noted as a breaking change, and our documentation would need to be updated. So we should just remember that.

**Comment by j9liu** [comment]
Thanks @kring! Merging now.

**Pull request merged**